### PR TITLE
Update phase1 (2017) tracking sequence

### PR DESCRIPTION
**Pull request by makortel**
This PR updates the Phase1 tracking sequence from the `Phase1PU70` to the one presented in tracking POG meeting Feb 1st. The performance comparison against `Phase1PU70` and run2 tracking (in phase1 detector) can be found from
https://indico.cern.ch/event/489234/contribution/5/attachments/1220760/1784567/slides_phase1.pdf
Due to the situation of the phase1 workflows (see the points below), the main purposes of this PR are to allow review/discussion on the implementation, and the use of the sequence e.g. for track selection MVA training.

Since the era migration of the reconstruction part of the phase1 customization is still pending, the new sequence is added following the current pattern (full configuration fragments + customize function). The customize for the Phase1PU70 is kept, and another customize for running run2 tracking in phase1 detector is added for reference. The tracking configuration is selected by changing which line in `phase1TkCustoms.py` is uncommented (not too handy, but hopefully good-enough while waiting for better solutions, see e.g. https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3597/1/2/1.html and onwards).

Also the effects on phase1 can't be easily tested before #12806 gets merged one way or another, and MultiTrackValidator configuration gets additional tuning for the iterations (that depends on #12806 and hence is not included in here).

Tested in 8_0_0_pre5. No changes expected in run2 workflows, for phase1 `-w upgrade -l 10000` runs.

@rovere @VinInn

### CODE REVIEW

**Comment by VinInn** [comment]
@cmsbuild, please test

**Comment by slava77** [comment]
@makortel  @VinInn @rovere 
What are the expectations for this PR with respect to integration in CMSSW, considering the era plans.
It sounds like this setup will be scrapped on a time scale of one pre-release or so.
If that's the case, should this stay in "RFC" (request for comments) state and be closed so that it does not stand in line for the imminent integration in CMSSW?

File: SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py [hunk 1]
```diff
@@ -427,3 +638,32 @@ def customise_Reco(process,pileup):
     process.preDuplicateMergingDisplacedTracks.trackProducers.remove("muonSeededTracksInOut")
 
     return process
+
+def customise_Reco_Run2(process):
```
Comment (slava77) [inline SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py]: Run2 spans move to Phase1. So, better not introduce much confusion. Maybe _Run2_Legacy and Run2_Phase1

**Comment by makortel** [comment]
@slava77 I will close this one for now. Additional reasons include
* cut-based selectors of highPtTripletStep and lowPtTripletStep need to be fixed because of #13055 (need to add `dr_exp` and `dz_exp` parameters)
* it's probably better to update the iterations in the validation sequences too (now that #12806 is in)

I can wait for the era plan to be finalized (let's hope it won't take too long) before the integration, as that way there is less moving customizations back and forth. @VinInn @rovere what do you think?

Anyway, I won't touch this branch, so it remains possible to run this version of phase1 tracking on 800pre5 (but no later) if anybody wants/needs.

**Pull request closed without merging**